### PR TITLE
stream raster bands

### PR DESCRIPTION
**Pull request by michaelmattig**
- [ ] I added an entry to [`CHANGELOG.md`](CHANGELOG.md) if knowledge of this change could be valuable to users.

---

Here is a brief summary of what I did:

<TEXT>

### CODE REVIEW

**Comment by michaelmattig** [comment]
[Man kann in Python mit Multiband-Rastern arbeiten](https://trello.com/c/NYpFDvcG/1579-man-kann-in-python-mit-multiband-rastern-arbeiten)

File: services/src/api/model/datatypes.rs [hunk 1]
```diff
@@ -992,11 +991,31 @@ impl From<QueryRectangle<SpatialPartition2D>>
             spatial_bounds: value.spatial_bounds.into(),
             time_interval: value.time_interval.into(),
             spatial_resolution: value.spatial_resolution.into(),
-            attributes: BandSelection::first(), // TODO: adjust once API supports attribute selection
+            attributes: geoengine_datatypes::primitives::BandSelection::first(), // TODO: adjust once API supports attribute selection
```
Comment (ChristianBeilschmidt) [inline services/src/api/model/datatypes.rs:994]: try_into and remove comment?

File: services/src/api/model/datatypes.rs [hunk 2]
```diff
@@ -1567,7 +1586,7 @@ pub enum RasterColorizer {
 impl RasterColorizer {
     pub fn band_selection(&self) -> BandSelection {
         match self {
-            RasterColorizer::SingleBand { band, .. } => BandSelection::new_single(*band),
+            RasterColorizer::SingleBand { band, .. } => BandSelection(vec![*band as usize]),
```
Comment (ChristianBeilschmidt) [inline services/src/api/model/datatypes.rs:1589]: why did you remove the constructor?

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: services/src/api/model/datatypes.rs (hunk 1 above)
Comment (michaelmattig) [reply services/src/api/model/datatypes.rs:994]: no, the comment is about the fact, that the `QueryRectangle` in the API does not have the `attributes` yet, because they are not yet used by clients. That's why it always select the first band here instead.

File: services/src/api/model/datatypes.rs (hunk 2 above)
Comment (michaelmattig) [reply services/src/api/model/datatypes.rs:1589]: I didn't remove the constructur but the type is now from `api` and not from `datatype`. The former one just doesn't have this method because it's only created by deserialization of user data and not from code.

**Review by ChristianBeilschmidt — APPROVED** [review]
